### ISSUE TITLE
Webpack4 TypeError: Cannot read property 'tap' of undefined

### ISSUE THREAD

**Issue by theskillwithin** [webpack-contrib/uglifyjs-webpack-plugin#240]
I think tap is new in webpak 4 "Replaced plugin calls with tap calls (new plugin system)"

```
➜  v2 git:(blue/webpack4) ✗ npm start

> mojo-marketplace-v2@2.0.0 start /Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2
> webpack --progress --watch --config webpack.config.babel.js

/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/uglifyjs-webpack-plugin/dist/index.js:267
            compilation.hooks.buildMobule.tap(plugin, buildMobuleFn);
                                          ^

TypeError: Cannot read property 'tap' of undefined
    at /Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/uglifyjs-webpack-plugin/dist/index.js:267:43
    at SyncHook.eval (eval at create (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/tapable/lib/HookCodeFactory.js:17:12), <anonymous>:189:1)
    at SyncHook.lazyCompileHook [as _call] (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/tapable/lib/Hook.js:35:21)
    at Compiler.newCompilation (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/webpack/lib/Compiler.js:407:26)
    at hooks.beforeCompile.callAsync.err (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/webpack/lib/Compiler.js:439:29)
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/tapable/lib/HookCodeFactory.js:24:12), <anonymous>:6:1)
    at AsyncSeriesHook.lazyCompileHook [as _callAsync] (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/tapable/lib/Hook.js:35:21)
    at Compiler.compile (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/webpack/lib/Compiler.js:434:28)
    at compiler.hooks.watchRun.callAsync.err (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/webpack/lib/Watching.js:76:18)
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/tapable/lib/HookCodeFactory.js:24:12), <anonymous>:6:1)
    at AsyncSeriesHook.lazyCompileHook [as _callAsync] (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/tapable/lib/Hook.js:35:21)
    at Watching._go (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/webpack/lib/Watching.js:39:32)
    at Watching.compiler.readRecords.err (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/webpack/lib/Watching.js:31:9)
    at Compiler.readRecords (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/webpack/lib/Compiler.js:331:11)
    at new Watching (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/webpack/lib/Watching.js:28:17)
    at Compiler.watch (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/webpack/lib/Compiler.js:161:10)
    at processOptions (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/webpack-cli/bin/webpack.js:508:14)
    at yargs.parse (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/webpack-cli/bin/webpack.js:516:3)
    at Object.parse (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/webpack-cli/node_modules/yargs/yargs.js:539:18)
    at /Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/webpack-cli/bin/webpack.js:211:8
    at Object.<anonymous> (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/webpack-cli/bin/webpack.js:518:3)
    at Module._compile (module.js:660:30)
    at Object.Module._extensions..js (module.js:671:10)
    at Module.load (module.js:573:32)
    at tryModuleLoad (module.js:513:12)
    at Function.Module._load (module.js:505:3)
    at Module.require (module.js:604:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/webpack/bin/webpack.js:12:2)
    at Module._compile (module.js:660:30)
    at Object.Module._extensions..js (module.js:671:10)
    at Module.load (module.js:573:32)
    at tryModuleLoad (module.js:513:12)
    at Function.Module._load (module.js:505:3)
    at Function.Module.runMain (module.js:701:10)
    at startup (bootstrap_node.js:190:16)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! mojo-marketplace-v2@2.0.0 start: `webpack --progress --watch --config webpack.config.babel.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the mojo-marketplace-v2@2.0.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/aupeterson/.npm/_logs/2018-02-24T00_29_16_700Z-debug.log
```

**Comment by michael-ciniawsky** [comment]
Fix coming in a few minutes thx for the report. `buildMobule`... :D

**Comment by michael-ciniawsky** [comment]
Released in [`v1.2.2`](https://github.com/webpack-contrib/uglifyjs-webpack-plugin/releases/tag/v1.2.2) 🎉
